src/components/networks/networkOverviewTab.jsx

```diff
@@ -149,3 +201,93 @@ export const NetworkOverviewTab = ({ network }) => {
+    const [ipAddress, setIpAddress] = useState('');
+    const [macAddress, setMacAddress] = useState('');
+    const [error, setError] = useState();
+
+    const add = () => {
+        setIsSubmitted(true);
+
+        if (!ipAddress || !macAddress)
+            return;
+
+        return networkAddStaticHostEntries({
+            connectionName: network.connectionName,
+            networkName: network.name,
+            macAddress,
+            ipAddress,
+            parentIndex,
+        })
+                .then(
+                    () => {
+                        setIsOpen(false);
+                        setIpAddress('');
+                        setMacAddress('');
+                        setError(undefined);
+                        networkGet({ connectionName: network.connectionName, id: network.id, updateOnly: true });
+                    },
+                    exc => setError(exc.message)
+                );
+    };
+
+    let addEntry = (
+        <Button variant="link"
+                id={idPrefix + "-static-host-entries-add"}
+                isDisabled={network.active}
+                isInline
+                onClick={() => setIsOpen(true)}>
+            {_("add entry")}
+        </Button>
+    );
+    if (network.active) {
+        addEntry = (
+            <Tooltip content={_("Deactivate the network in order to edit it")}>
+                {addEntry}
+            </Tooltip>
+        );
+    }
+
+    return (
+        <>
+            {addEntry}
+            <Modal position="top" variant="small" isOpen={isOpen} onClose={() => setIsOpen(false)}
+                title={_("Add a DHCP static host entry")}
+                footer={
+                    <>
+                        {error && <ModalError dialogError={_("Failed to save network settings")} dialogErrorDetail={error} />}
+                        <Button variant='secondary' onClick={add} id="add-new-static-entry-save">
+                            {_("Add")}
+                        </Button>
+                        <Button variant='link' id="add-new-static-entry-cancel" onClick={() => setIsOpen(false)}>
+                            {_("Cancel")}
+                        </Button>
+                    </>
+                }>
+                <Form onSubmit={e => {
+                    e.preventDefault();
+                }} isHorizontal>
+                    <FormGroup label={_("MAC address")} fieldId="add-new-static-entry-mac-address"
+                               helperTextInvalid={_("MAC address must not be empty")}
+                               validated={isSubmitted && !macAddress ? "error" : "default"}>
+                        <TextInput id="add-new-static-entry-mac-address"
+                                   validated={isSubmitted && !macAddress ? "error" : "default"}
+                                   value={macAddress}
+                                   onChange={setMacAddress} />
+                    </FormGroup>
```

Comment:
As mentioned in the comment below, There is no such thing as MAC address for IPv6. So when defining Static entries for IPv6, user should instead input "name", or alternatively "id".

Reply:
Removed feature for iPV6